shell/platform/android/test/io/flutter/embedding/android/FlutterViewTest.java

```diff
@@ -386,6 +390,29 @@ public void flutterImageView_acquiresImageAndInvalidates() {
     verify(imageView, times(1)).invalidate();
   }
 
+  @Test
+  public void flutterImageView_acquiresMaxImagesAtMost() {
+    final ImageReader mockReader = mock(ImageReader.class);
+    when(mockReader.getMaxImages()).thenReturn(2);
+
+    final Image mockImage = mock(Image.class);
+    when(mockReader.acquireLatestImage()).thenReturn(mockImage);
+
+    final FlutterImageView imageView =
+        spy(
+            new FlutterImageView(
+                RuntimeEnvironment.application,
+                mockReader,
+                FlutterImageView.SurfaceKind.background));
+
+    doNothing().when(imageView).invalidate();
```

Comment:
Added this test case. `invalidate()` doesn't provide guarantees on when `onDraw(Canvas canvas)` is called.

https://developer.android.com/reference/android/view/View#invalidate()
> Invalidate the whole view. If the view is visible, onDraw(android.graphics.Canvas) will be called at some point in the future.